qualtran/_infra/bloq_example.py

```diff
@@ -69,6 +70,18 @@ def _bloq_cls_from_func_annotation(func: Callable[[], Bloq]) -> Type[Bloq]:
     return cls
 
 
+@typing.overload
```

Comment:
Question: What are these new overloads for?

Reply:
this is a type annotation thing. It lets you be more precise about the annotations in two different regimes rather than just `Union`ing all the types

Reply:
https://docs.python.org/3/library/typing.html#overload